### ISSUE TITLE
Removed builders show up in status displays

### ISSUE THREAD

**Issue by tardyp** [buildbot/buildbot#2959]
This ticket is a migrated Trac ticket [3386](http://trac.buildbot.net/ticket/3386)

People contributed to the original ticket: `verm@...`
Ticket created on: `Dec 08 2015`
Ticket last modified on: `Dec 08 2015`

---

If you remove a builder it will still exist in the status display.

This is a huge problem when initially setting up as you may be renaming a lot.

It is also an issue for down-the-road maintenance where builder names can change to reflect new configurations.

---

**Comment by seankelly** [comment]
I think old builders are hidden by default now.

**Comment by tichy3000** [comment]
The old builders still show up at least in the Waterfall View. There is no "Show old builders" checkbox like for the Builders page. (Using version 0.9.14)

**Comment by giganteous** [comment]
Hi,
This ticket is still open.

I'd suggest not fixing it, or changing the milestone, or document how to remove old cruft from the state database by hand (for the experts :-). A friendly suggestion, thanks!

**Comment by TTimo** [comment]
There is a 'Show old builders' page, but a way to clean out old builders would still be quite useful.

**Comment by krf** [comment]
Somewhat related: https://github.com/buildbot/buildbot/issues/4327 

I'd really like to see this being implemented as well. Our database keeps growing, as we change the builder names frequently.

**Comment by thendrix** [comment]
Now with 2.x not using pickles, and being 100% SQL database is there is preferred method to purge the builder out of the database?  I would assume this means using sqlite3 to open the database and nuke the tables would be the only way for now...  seems safe and easy for the average joe.

```
sqlite> SELECT * FROM builders;
id          name        description  name_hash                               
----------  ----------  -----------  ----------------------------------------
1           runtests                 ae3c03f408678b20c1eb105fc44bf038def161a9
```

**Comment by tardyp** [comment]
@thendrix indeed, deleting in the database is the option. There is a pending PR about it from @rjarry 
https://github.com/buildbot/buildbot/pull/4554

Also, if you mean to grow your CI, it is a good idea to use postgres from the start.
as 2.x is using extensively the db, sqlite starts to show its limit.

**Comment by ryandesign** [comment]
> There is a pending PR

The PR was closed without merging it. So what's the status of this issue now?

**Comment by mojca** [comment]
I'm not aware of any fix being deployed, and despite updating the versions frequently (maybe I don't have the latest one), we are still bitten by this problem:
    https://build.contextgarden.net/#/waterfall

OpenBSD 6.5 is gone, for example. But at the moment it's impossible to see whether a particular builder is gone for good, offline (and needs an urgent action), or just didn't get anything to build for a while.

(0.8 was better in that respect.)

At some point I was fed up with a large number of defunct/obsolete builders cluttering the overview, so I deleted the database and started from scratch, as I don't need to worry about the history in any way. But this in not a general solution.

**Comment by thendrix** [comment]
My "fix" suggestion was just to manually edit the database for each instance, but like I said that's not a real solution for the average end user.

**Comment by hmoffatt** [comment]
Although the builders page can hide old builders now, the schedulers page still lists all schedulers. Is there an easy way to remove the ones that have no master?
